docs/migrations/v4-to-v5.md

```diff
@@ -0,0 +1,390 @@
+# Migrating from v4 to v5
+
+This document is intended to highlight significant changes which may require more information than presented in the CHANGELOG.
+Any changes that must be done by a user of ibc-go should be documented here.
```

Comment:
I think we can probably remove this section going forward. I'd assume it is just noise nowadays